src/main/scala/io/qbeast/spark/index/query/QueryExecutor.scala

```diff
@@ -19,23 +19,25 @@ class QueryExecutor(querySpecBuilder: QuerySpecBuilder, qbeastSnapshot: QbeastSn
    * Executes the query on each revision according to their QuerySpec
    * @return the final sequence of blocks that match the query
    */
-  def execute(): Seq[QbeastBlock] = {
+  def execute(): Iterable[QbeastBlock] = {
```

Comment:
Why this change? The function returns a Seq type, could it lead to issues?

Reply:
We need the sequence of blocks to be unique. Since OR's queries are a union of queries, it could lead to duplicated blocks. Iterable is a more flexible interface to implement it.

Reply:
Don't think it could lead to issues. Seq is an Iterable object.

Reply:
Okay, thanks for the explanation